server/asset/btc/btc.go

```diff
@@ -573,6 +573,10 @@ func ParseBondTx(ver uint16, rawTx []byte, chainParams *chaincfg.Params) (bondCo
 		err = fmt.Errorf("bad bond pkScript")
 		return
 	}
+	if !segwit && len(scriptHash) == 32 {
+		err = fmt.Errorf("%s backend does not support segwit bonds", chainParams.Name)
```

Comment:
Maybe the strictness is ultimately pointless, but I was thinking for a given bondver and assetID, we should expect either segwit or not segwit for consistency and to keep it standardized for a given asset.

```go
switch len(scriptHash) {
case 32:
// require segwit
case 20:
// require !segwit
}
```